### ISSUE TITLE
Lambda Layers Import Error

### ISSUE THREAD

**Issue by gamalalice** [aws/chalice#1360]
Hello, I am trying to use the Chalice layers technique. I have a chalice package that will exceed 50MB if I add the `aiohttp` library. So layers route is the way to go. 

I have read each of the other issues tagged with a layers error.
Create a requirements-dev.txt file with all of the packages and do not add to requirements.txt file. 
In this case I only learned about layers after i was 3 months into my project, so i always had a requirements.txt file with everything. So, what I have done is make a copy of the requirements.txt file in the requirements-dev.txt file with everything -- including aiohttp. Then I have the requirements.txt file which has everything but the aiohttp library. I add the layers to my config.json, but in the end it says ImportError. I have no clue what I am doing wrong. 

**requirements.txt file**
```
argh==0.26.2
attrs==19.3.0
bcrypt==3.1.7
boto3==1.10.10
botocore==1.13.10
cairocffi==1.1.0
CairoSVG==2.4.2
certifi==2019.9.11
cffi==1.13.2
chardet==3.0.4
Click==7.0
colorama==0.4.3
cssselect2==0.2.2
defusedxml==0.6.0
docutils==0.15.2
entrypoints==0.3
enum-compat==0.0.3
flake8==3.7.9
flake8-polyfill==1.0.2
future==0.18.2
idna==2.8
imgkit==1.0.2
jmespath==0.9.4
lxml==4.4.1
mando==0.6.4
MarkupSafe==1.1.1
mccabe==0.6.1
multidict==4.7.4
numpy==1.17.3
pandas==0.25.1
pathtools==0.1.2
Pillow==6.2.1
pycodestyle==2.5.0
pycparser==2.19
pyflakes==2.1.1
PyPDF2==1.26.0
python-dateutil==2.8.0
python-dotenv==0.10.5
pytz==2019.3
PyYAML==5.3
radon==4.0.0
requests==2.22.0
s3transfer==0.2.1
six==1.12.0
stripe==2.0.1
tinycss2==1.0.2
urllib3==1.25.6
webencodings==0.5.1
yarl==1.4.2
```

**requirements-dev.txt**
```
aiohttp==3.6.2
argh==0.26.2
attrs==19.3.0
bcrypt==3.1.7
boto3==1.10.10
botocore==1.13.10
cairocffi==1.1.0
CairoSVG==2.4.2
certifi==2019.9.11
cffi==1.13.2
chardet==3.0.4
Click==7.0
colorama==0.4.3
cssselect2==0.2.2
defusedxml==0.6.0
docutils==0.15.2
entrypoints==0.3
enum-compat==0.0.3
flake8==3.7.9
flake8-polyfill==1.0.2
future==0.18.2
idna==2.8
imgkit==1.0.2
jmespath==0.9.4
lxml==4.4.1
mando==0.6.4
MarkupSafe==1.1.1
mccabe==0.6.1
multidict==4.7.4
numpy==1.17.3
pandas==0.25.1
pathtools==0.1.2
Pillow==6.2.1
pycodestyle==2.5.0
pycparser==2.19
pyflakes==2.1.1
PyPDF2==1.26.0
python-dateutil==2.8.0
python-dotenv==0.10.5
pytz==2019.3
PyYAML==5.3
radon==4.0.0
requests==2.22.0
s3transfer==0.2.1
six==1.12.0
stripe==2.0.1
tinycss2==1.0.2
urllib3==1.25.6
webencodings==0.5.1
yarl==1.4.2
```

```
{
  "version": "2.0",
  "app_name": "***",
  "environment_variables": {***},
  "stages": {
    "dev": {
      "api_gateway_stage": "api",
      "subnet_ids": ["***"],
      "security_group_ids": ["***"],
      "layers": ["aiohttp:2"],
      "lambda_timeout": 30,
      "lambda_memory_size": 640,
      "environment_variables": {***},
      "lambda_functions": {*** }
    },
    "test": {
      "api_gateway_stage": "api",
      "subnet_ids": ["***"],
      "security_group_ids": ["***"],
      "layers": ["aiohttp:2"],
      "lambda_timeout": 30,
      "lambda_memory_size": 640,
      "environment_variables": {***},
      "lambda_functions": {***}
      "environment_variables": {***},
      "lambda_functions": {
        "create_postcards": {
          "lambda_memory_size": 3008
        }
      }
    },
    "dev-test": {
      "api_gateway_stage": "api",
      "subnet_ids": ["***"],
      "security_group_ids": ["***"],
      "layers": ["aiohttp:2"],
      "lambda_timeout": 30,
      "lambda_memory_size": 640,
      "environment_variables": {***},
      "lambda_functions": {*** }
      "environment_variables": {***},
      "lambda_functions": {
        "create_postcards": {
          "lambda_memory_size": 3008
        } 
      }
    }
  }
}
```

So, I want to deploy dev-test, but every single time I get an import error. I simply cannot understand what I could possibly be doing wrong. 

chalice 1.11.0 python 3.7.6

**Comment by jamesls** [comment]
Is this the real value you're using in your config.json or is it a sanitized version?

```
"layers": ["aiohttp:2"],
```

The `layers` key needs to be a list of layer ARNs, so it should look something like `["arn:aws:..."]`.  Also, how are you creating your layer?

**Comment by gamalalice** [comment]
Hello, sorry. 

I didn't see this until now. Damnit. 

I am using the arn value, I just wanted to avoid throwing it out on the world wide web. 
When I create the layer, I download aiohttp on my local computer and all of its friends, zip it, place it up in aws lambda layer, and reference the arn in my config.json file

**Comment by jamesls** [comment]
So I would first double check that the layer is being properly added to your function.  For example, you can run:

```
aws lambda get-function-configuration --function-name $APP-$STAGE-$FUNCTION
```

so for example, if my app is named "foo" then the function name would be `foo-dev-create_postcards`.

You should see a `Layers` key with the ARN that you specified.  If you aren't seeing that, then it means the layer isn't being added to your lambda function.

If you *do* see the layer configuration you expect then that suggests it's an issue with how you're building your layer.

Let me know what you see when you run that command and I can help you further.

**Comment by gamalalice** [comment]
Hey, so i ran the command and I can confirm that I am seeing the layers with the ARN. Here is the output:

**Comment by gamalalice** [comment]
Hey James, 

How are you. Your point:

"If you do see the layer configuration you expect then that suggests it's an issue with how you're building your layer."

I decided to research more and learned that there is a specific package structure that is important. Before I had been uploading only the aiohttp folder as a zip. I see now that it is required to upload it as `python/lib/python3.7/site-packages/aiohttp`

I tried it and it worked. 

Thank you for taking the time.
